packages/grid/_modules_/grid/hooks/features/selection/useGridSelection.ts

```diff
@@ -136,29 +167,43 @@ export const useGridSelection = (apiRef: GridApiRef, props: GridComponentProps):
 
       const hasCtrlKey = event.metaKey || event.ctrlKey;
 
-      // Without checkboxSelection, multiple selection is only allowed if CTRL is pressed
-      const isMultipleSelectionDisabled = !checkboxSelection && !hasCtrlKey;
-      const resetSelection = !canHaveMultipleSelection || isMultipleSelectionDisabled;
-
-      if (resetSelection) {
-        apiRef.current.selectRow(
-          params.id,
-          hasCtrlKey || checkboxSelection ? !apiRef.current.isRowSelected(params.id) : true,
-          true,
+      if (canHaveMultipleSelection && event.shiftKey) {
+        apiRef.current.selectRowRange(
+          {
+            startId: lastRowToggledByClick.current ?? params.id,
+            endId: params.id,
+          },
+          !apiRef.current.isRowSelected(params.id),
         );
+
+        event.stopPropagation();
```

Comment:
Was it added to prevent the row highlighting during the click? I believe that with the [Selection API](https://developer.mozilla.org/en-US/docs/Web/API/Selection/removeAllRanges) we can do that.

Reply:
`removeAllRanges` works great if applied on `cellKeyDown`
If applied on `rowClick`, we have a selection blink during the click.

![image](https://user-images.githubusercontent.com/3309670/133037030-f1a1fb43-1c28-48ce-8bb8-b77630472c2b.png)

The compatibility seems quite good

Reply:
See https://github.com/mui-org/material-ui-x/pull/2456/commits/89d1567649254fd73dc476c7b2ceace23d2e6a1c